src/containers/LocationPage/LocationPage.jsx

```diff
@@ -641,14 +643,33 @@ class LocationPage extends PureComponent {
     );
   }
 
+  renderLocationSearch() {
+    return (
+      <LocationSearch />
```

Comment:
The problem with using the default callback is that you will lose your filter settings. Maybe you should add a custom handler that does queryRebuild to maintain them?